apps/browser/lib/hooks/useEvoService.ts

```diff
@@ -0,0 +1,184 @@
+import {
+  allowTelemetryAtom,
+  capReachedAtom,
+  evoServiceAtom,
+  localOpenAiApiKeyAtom,
+  showAccountModalAtom,
+  userWorkspaceAtom,
+  errorAtom
+} from "@/lib/store";
+import { useCreateChat } from "@/lib/mutations/useCreateChat";
+import { useAddChatLog } from "@/lib/mutations/useAddChatLog";
+import { useAddMessages } from "@/lib/mutations/useAddMessages";
+import { useChats } from "@/lib/queries/useChats";
+import { useAddVariable } from "@/lib/mutations/useAddVariable";
+import { useUpdateUserFiles } from "@/lib/hooks/useUpdateUserFiles";
+import { ChatLog } from "@/components/Chat";
+import { EvoThreadCallbacks, EvoThreadConfig } from "@/lib/services/evo/EvoThread";
+import { v4 as uuid } from "uuid";
+import { useAtom } from "jotai";
+import { useState, useEffect, useRef } from "react";
+import { Workspace, InMemoryWorkspace } from "@evo-ninja/agent-utils";
+import { ChatLogType, ChatMessage } from "@evo-ninja/agents";
+
+export const useEvoService = (
+  chatId: string | "<anon>" | undefined,
+  isAuthenticated: boolean,
```

Comment:
Why are we specifically not using the atoms for these but rather requiring them as props?

Reply:
Same response as above, figured it was simpler to have a flag passed in,  than to use the `useSession` hook w/ an added condition of `session === "authenticated"`. Here you can just do `isAuthenticated` and not have to worry about the `next/auth` library we're using in the root application. Limits dependency spread.